python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -625,6 +625,7 @@ class AutoMaterializeAssetEvaluation(NamedTuple):
     num_skipped: int
     num_discarded: int
     run_ids: Set[str] = set()
+    rule_snapshots: Optional[Sequence[AutoMaterializeRuleSnapshot]] = list()
```

Comment:
Allow this field to be optional to be able to deserialize old asset evaluations where this field was not populated

Reply:
nit: maybe default this to None? doesn't really make a difference but makes it easier to make sure the path of having it set to None works properly

Reply:
good call